Userland/Utilities/acpi.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2022-2022, Andreas Kling <kling@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibCore/ArgsParser.h>
+#include <LibCore/File.h>
+#include <LibCore/System.h>
+#include <stdio.h>
+#include <unistd.h>
+
+// https://uefi.org/specs/ACPI/6.4/05_ACPI_Software_Programming_Model/ACPI_Software_Programming_Model.html#system-description-table-header
+struct [[gnu::packed]] SDTHeader {
+    char sig[4];
+    u32 length;
+    u8 revision;
+    u8 checksum;
+    char oem_id[6];
+    char oem_table_id[8];
+    u32 oem_revision;
+    u32 creator_id;
+    u32 creator_revision;
+};
+
+void pkgLength(Vector<u8> const& data, u32& next_in_block, u32& next_block);
+
+void pkgLength(Vector<u8> const& data, u32& next_in_block, u32& next_block)
+{
+    u32 pos = next_block;
+
+    outln("");
+    out("data: ");
+    if (pos > 3) {
+        for (auto i = -3; i < 0; i++)
+            out("{:#02x} ", data.at(pos + i));
+    } else
+        out("               ");
+    out("  ");
+    for (auto i = 0; i < 5; i++)
+        out("{:#02x} ", data.at(pos + i));
+    outln("");
+
+    u32 length = 0;
+    u32 delta = 0;
+    int const size = data.at(pos + 1) >> 6;
+    if (size == 0) {
+        length = data.at(pos + 1) & 0x3f;
+        delta = 1;
+    } else if (size == 1) {
+        length = data.at(pos + 1) & 0x0f;
+        length = length | (data.at(pos + 2) << 4);
+        delta = 2;
+    } else if (size == 2) {
+        length = data.at(pos + 1) & 0x0f;
+        length = length | (data.at(pos + 2) << 4);
+        length = length | (data.at(pos + 3) << 12);
+        delta = 3;
+    } else {
+        length = data.at(pos + 1) & 0x0f;
+        length = length | (data.at(pos + 2) << 4);
+        length = length | (data.at(pos + 3) << 12);
+        length = length | (data.at(pos + 4) << 20);
+        delta = 4;
+    }
+    length -= (delta - 1);
+
+    outln("pos: {} {} {}", delta, pos + delta + length, data.size());
+    next_in_block = next_block + 1 + delta;
+    next_block = next_block + delta + length;
+}
+
+ErrorOr<int> serenity_main(Main::Arguments arguments)
+{
+    TRY(Core::System::pledge("stdio rpath"));
+
+    StringView filename;
+
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(filename, "Name of DSDT table", "DSDT table", Core::ArgsParser::Required::No);
+    args_parser.parse(arguments);
+
+    int fd;
```

Comment:
Please use `Core::Stream::File`
You might find [this](https://github.com/SerenityOS/serenity/blob/a66dbef1ed7c4159207e0f28fa85d5f305f08582/Userland/Utilities/cmp.cpp#L13-L22) interesting